_posts/2021-07-05-pcp-grafana.md

```diff
@@ -0,0 +1,106 @@
+
+```
+# rpm -q cockpit
+cockpit-247-1.fc34.x86_64
+# systemctl enable --now cockpit.socket
+# firewall-cmd --permanent --add-service cockpit
+# firewall-cmd --reload
+```
+
+## Metrics in Cockpit for the local machine
+
+Then you can log into Cockpit on [https://localhost:9090](https://localhost:9090), or the corresponding host name of the system which you want to troubleshoot. The Overview page shows a very brief CPU/memory usage with a "View Details and history" link:
+
+![Overview usage card](/images/overview-card-usage.png)
+
+That link leads you to the Metrics page, which in a minimal installation only shows current metrics:
+
+![Metrics page without PCP](/images/metrics-page-no-pcp.png)
+
+For spotting performance problems in the past, we need to enable historical metrics, which are provided by [Performance Co-Pilot](https://pcp.io/), aka.  "PCP".
+
+The "Install cockpit-pcp" button installs the necessary support for this, and asks you to log in again (the new cockpit-pcp bridge cannot be used in an existing session). After that, the Metrics page will start to show historical metrics. Combined with Cockpit's other pages such as "Services", "Logs", or "Podman containers", this is a very powerful tool to find performance related problems:
+
+<iframe width="960" height="720" src="https://www.youtube.com/embed/8_NLoJBjay4?rel=0" frameborder="0" allowfullscreen></iframe>
+
+
+## Metrics of multiple machines
+
+But what if you want to monitor not just one machine, but your whole fleet?
+
+Cockpit used to have a "Dashboard" to see some basic resource usage of up to twenty machines, but this was very inflexible, inefficient, insecure, lacking important features such as alerting, and was bound to a running Cockpit session. That's why the Dashboard got removed in [Cockpit 234](https://cockpit-project.org/blog/cockpit-234.html).
+
+One of today's industry standard solutions for visualizing metrics is [Grafana](https://grafana.com/); it does not have above limitations. PCP offers a [Grafana plugin](https://grafana-pcp.readthedocs.io/en/latest/) to provide the machine's metrics to a local or remote Grafana installation.
+
+## Setup of monitored machines
+
+Setting up the machines for sending metrics to Grafana involves a couple of steps:
+
+ * Install PCP and enable `pmlogger.service` to start data collection. This already happened with the "Install cockpit-pcp" button above.
+ * Enable `pmproxy.service`, the PCP component which exposes a read-only metrics query API to the network.
+ * pmproxy needs the [Redis database](https://redis.io/), so the `redis` package needs to be installed, and the service configured to start with `pmproxy.service`.
+ * pmproxy opens a network port (44322 by default), which needs to be opened in the firewall for the zone that contains the Grafana machine.
```

Comment:
While I'm reading this entire section, at each separate point, I'm asking myself "...uh.. but how do I do that?  Where's the CLI fragment like was provided above for installing cockpit?  Do I have to figure this out on my own?  Should I open a terminal now and poke around?  Try Google?"

...of course, the answer to all of these questions is just below, but it might save the user some frustration to put it up top and use this list to describe all of the things that will happen as a result of the magic button.

Reply:
Good point, thanks for bringing it up! I restructured this paragraph a bit to make this obvious from the start. WDYT?

Reply:
That helps a lot, thanks :)